pandas/core/groupby/generic.py

```diff
@@ -88,6 +90,9 @@
 
 from pandas.plotting import boxplot_frame_groupby
 
+if TYPE_CHECKING:
+    from pandas._libs import Interval
```

Comment:
inside TYPE_CHECKING blocks can safely import from the top level without causing circular imports.

 ```suggestion
     from pandas import Interval
```